pyinfra/facts/apt.py

```diff
@@ -7,6 +7,23 @@
 from .gpg import GpgFactBase
 from .util import make_cat_files_command
 
+def noninteractive_apt(command: str, force=False):
+    args = ["DEBIAN_FRONTEND=noninteractive apt-get -y"]
+
+    if force:
+        args.append("--force-yes")
+
+    args.extend(
+        (
+            '-o Dpkg::Options::="--force-confdef"',
+            '-o Dpkg::Options::="--force-confold"',
+            command,
+        ),
+    )
+
+    return " ".join(args)
+
+APT_CHANGES_RE = re.compile(r"^(\d+) upgraded, (\d+) newly installed, (\d+) to remove and (\d+) not upgraded.$")
```

Comment:
Interesting! This is not an approach I'd thought of using at all, I like it.

Reply:
And this line should also be present when executing e.g. `apt-get autoremove`, hence my question https://matrix.to/#/!dPvLqxqwbQZYeQrNnu:matrix.org/$ygUK_H5ACvxHxmEbKiwfSBsCQsoiTINm9WqBl5tSatE?via=beeper.com&via=matrix.org&via=t2bot.io if there was a way for operations to generate a command and then decide, based on the output, if any change was actually performed.
That could then also apply to `apt-get update` and then almost all apt operations could support proper change detection.